middleware/middleware.go

```diff
@@ -0,0 +1,75 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/services/stats"
+)
+
+func MaxConcurrentRequestsMiddleware(maxConcurrentRequests int) func(http.Handler) http.Handler {
```

Comment:
[minor] More compact name, no need to include `Middleware` in the function name since it is on the package.

Think of usage `middleware.MaxConcurrentRequestsMiddleware( )`, it can be a bit excessive. 

```suggestion
func LimitConcurrentRequests(maxRequests int) func(http.Handler) http.Handler {
```

Reply:
done